packages/application-shell/src/components/gtm-booter/gtm-booter.js

```diff
@@ -16,28 +16,26 @@ export const GtmContext = React.createContext({
   getHierarchy: gtm.getHierarchy,
 });
 
-class GtmBooter extends React.Component {
-  static displayName = 'GtmBooter';
-  static propTypes = {
-    children: PropTypes.node.isRequired,
-    trackingEventWhitelist: PropTypes.objectOf(
-      PropTypes.oneOfType([
-        PropTypes.string,
-        PropTypes.objectOf(PropTypes.string),
-      ])
-    ).isRequired,
-  };
-  componentDidMount() {
+const GtmBooter = props => {
+  React.useEffect(() => {
     // We don't need any user data to start using GTM, for example for
     // tracking page views and flows when the user is not logged in.
     gtm.boot({
       ...defaultTrackingEventWhitelist,
-      ...this.props.trackingEventWhitelist,
+      ...props.trackingEventWhitelist,
     });
-  }
-  render() {
-    return this.props.children;
-  }
-}
+  }, [props.trackingEventWhitelist]);
```

Comment:
Note that this is actually a change in behaviour where boot will be called again if the white list changes. Whereas before it would not.

Reply:
Ah good point, let me remove it then